refactor: [RC] return error if scala db is missing

MohamadJaara opened the pull request:
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

there is the possibility for the migration to fail if the scala db cannot be accessed for any reason.

### Solutions

even tho such a failure have a very slim to non-chance of happening treating this error gracefully and returning an error is the correct way to do it 

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

### Testing

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 
<!-- Uncomment the brackets and place your screenshots on the table below. -->
<!--
| Before | After |
| ----------- | ------------ |
|   |   |
-->

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

someoneEsle commented:
build again

someoneEsle commented:
I'm not sure why, but wireless guests now appear as "Default" (I see system messages "default joined" and "default left"). Pretty sure this was not the case yesterday

deanrobertcook commented:
@someoneEsle this is a more general issue - made an issue here we can include in a further sprint:
https://github.com/wireapp/android-project/issues/24

MohamadJaara committed:
refactor: return error if scala user DB is missing

MohamadJaara committed:
refactor: return error if scala user DB is missing
Merge remote-tracking branch 'origin/release/candidate' into refactor/return_error_if_scala_DB_is_missing

# Conflicts:
#	app/src/main/kotlin/com/wire/android/migration/feature/MigrateMessagesUseCase.kt

AndroidBob commented:
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1482/1/) **failed**.

MohamadJaara committed:
tests

MohamadJaara committed:
Merge branch 'release/candidate' into refactor/return_error_if_scala_DB_is_missing

AndroidBob commented:
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1482/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.1(5809449)(5809449)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.1(5809449)(5809449)-staging-compat.apk)

MohamadJaara committed:
Merge branch 'release/candidate' into refactor/return_error_if_scala_DB_is_missing

AndroidBob commented:
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1482/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.1(5810548)(5810548)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.1(5810548)(5810548)-staging-compat.apk)

MohamadJaara committed:
Merge remote-tracking branch 'origin/release/candidate' into refactor/return_error_if_scala_DB_is_missing

# Conflicts:
#	app/src/main/kotlin/com/wire/android/migration/feature/MigrateUsersUseCase.kt

AndroidBob commented:
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1482/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.1(5810662)(5810662)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.1(5810662)(5810662)-staging-compat.apk)
